lapack/gonum/dtgsyl.go

```diff
@@ -86,11 +109,197 @@ func (impl Implementation) Dtgsyl(trans string, ijob int, m int, n int, a []floa
+				impl.Dlacpy('F', m, n, work, m, f, ldf)
+				scale = scale2
+			}
+			//cierro for
+		} //cierro for
+		//*retrun ????
+	}
+
+	//     Determine block structure of A
+
+	p = 0
+	i = 1
+G40: //CONTINUE
+	if i > m {
+		goto G50
+	}
+	p = p + 1
+	iwork[p] = float64(i)
+	i = i + mb
+	if i >= m {
+		goto G50
+	}
+	if a[i][i-1] != 0 {
+		i = i + 1
+	}
+	goto G40
+G50: //CONTINUE
+	iwork[p+1] = float64(m + 1)
+	if iwork[p] == iwork[p+1] {
+		p = p - 1
+	}
+
+	//     Determine block structure of B
+
+	q = p + 1
+	j = 1
+G60: //CONTINUE
+	if j > n {
+		goto G70
+	}
+	q = q + 1
+	iwork[q] = float64(j)
+	j = j + nb
+	if j >= n {
+		goto G70
+	}
+	if b[j][j-1] != 0 {
+		j = j + 1
+	}
+	goto G60
+G70: //CONTINUE
+	iwork[q+1] = float64(n + 1)
+	if iwork[q] == iwork[q+1] {
+		q = q - 1
+	}
+	if notran {
+
+		//DO 150 iround = 1, isolve
+		for iround := 1; iround <= isolve; iround++ {
+			//           Solve (I, J)-subsystem
+			//              A(I, I) * R(I, J) - L(I, J) * B(J, J) = C(I, J)
+			//              D(I, I) * R(I, J) - L(I, J) * E(J, J) = F(I, J)
+			//          for I = P, P - 1,..., 1; J = 1, 2,..., Q
+			dscale = 0
+			dsum = 1
+			pq = 0
+			scale = 1
+			//DO 130 j = p + 2, q
+			for j := (p + 2); j <= q; j++ {
+				js = int(iwork[j])
+				je = int(iwork[j+1] - 1)
+				nb = je - js + 1
+				//DO 120 i = p, 1, -1
+				for i := p; i <= 1; i-- {
+					is = int(iwork[i])
+					ie = int(iwork[i+1] - 1)
+					mb = ie - is + 1
+					ppqq = 0
+					//dtgsy2( trans, ifunc, mb, nb, a( is, is ), lda,b( js, js ), ldb, c( is, js ), ldc,d( is, is ), ldd, e( js, js ), lde,f( is, js ), ldf, scaloc, dsum, dscale,iwork( q+2 ), ppqq, linfo )
+					if linfo > 0 {
+						info = linfo
+					}
+					pq = pq + ppqq
+					if scaloc <= float64(1) {
+						//DO 80 k = 1, js - 1
+						for k := 1; k <= js; js-- {
+							//bi.Dscal(m, scaloc, c(1, k), 1)
+							//bi.Dscal(m, scaloc, c[k:], ldc) no lo toma tampoco
+							bi.Dscal(m, scaloc, f[k:], 1)
+							//80                CONTINUE
+						}
+						//DO 90 k = js, je
+						for k := js; k <= je; je++ {
+							//bi.Dscal(is-1, scaloc, c(1, k), 1)
+							bi.Dscal(is-1, scaloc, f[k:], 1)
+							//90                CONTINUE
+						}
+						//DO 100 k = js, je
+						for k := js; k <= js; je++ {
+							//bi.Dscal(m-ie, scaloc, c(ie+1, k), 1)
+							//bi.Dscal(m-ie, scaloc, f(ie+1, k), 1)
+							bi.Dscal(m-ie, scaloc, f[(ie+1)*ldc+k:], 1)
+							//100                CONTINUE
+						}
+						//DO 110 k = je + 1, n
+						for k := je + 1; k <= (je + 1); k++ {
+							//bi.Dscal(m, scaloc, c(1, k), 1)
+							//bi.Dscal(m, scaloc, f(1, k), 1)
+							bi.Dscal(m, scaloc, f[(ie+1)*ldc+k:], 1)
+							//110                CONTINUE
+						}
+						scale = scale * scaloc
+					}
+					//                 Substitute R(I, J) and L(I, J) into remaining
+					//                 equation.
+					if i > 1 {
+						//bi.Dgemm('N', 'N', is-1, nb, mb, -1, a[1][is] , lda, c[is][js], ldc, 1, c[1][js], ldc)
```

Comment:
* Usar tipos de `lapack/lapack.go`
* Argumento `m` (tercero de Dgemm) es variable longitud. Si se usa un indice (`is = iwork[i]`) se tiene que agregar 1, puede ser que se tenga que hacer lo mismo para `nb` y `mb` pero no estoy seguro a primera vista
* 
```fortran
CALL dgemm( 'N', 'N', is-1, nb, mb, -one,
      $                           a( 1, is ), lda, c( is, js ), ldc, one,
      $                           c( 1, js ), ldc )
```
=>
```go
bi.Dgemm(lapack.NoTrans, lapack.NoTrans, is, nb, mb, -1, a[is:] , lda, c[is*ldc+js:], ldc, 1, c[js:], ldc)
```